src/lib/auth/PacAuthenticator.ts

```diff
@@ -0,0 +1,20 @@
+import {
+    ClientCredentials,
+    CdsEnvironment,
+    UsernamePassword,
+} from "@microsoft/powerplatform-cli-wrapper";
+
+export interface PacAuthenticator {
```

Comment:
I assume this interface and its related siblings will eventually move to the cli-wrapper package? Those seem agnostic to GH vs AzDO... right now, things appear unexpected, but this might just be a point in time thing? We can discuss offline if necessary

Reply:
TL;DR this is a point in time thing.

PacAuthenticator is intended as a barrier between the pp-actions current AuthHandler class and the cli-wrapper. AuthHandler is now injected with a PacAuthenticator, which will take whatever AuthType / AuthKind combination is selected and execute the appropriate methods on the runner it represents (whether it's the legacy runner currently in this codebase or the runner from the CLI wrapper).

Eventually, once we've phased out the legacy runners, we can remove PacAuthenticator if we want to, but we've effectively taken one responsibility out of AuthHandler and moved it into its dependency. AuthHandler's job is now to determine the appropriate AuthType / AuthKind, and PacAuthenticator executes that method.